drivers/npmx/npmx_shell.c

```diff
@@ -570,6 +570,83 @@ static int cmd_charger_trickle_set(const struct shell *shell, size_t argc, char
 	return 0;
 }
 
+static int cmd_charger_discharging_current_get(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
+	uint16_t current;
+	npmx_error_t err_code = npmx_charger_discharging_current_get(charger_instance, &current);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Value: %d mA.", (int)current);
+	} else {
+		shell_error(shell, "Error: unable to read trickle voltage value.");
+	}
+
+	return 0;
+}
+
+static int cmd_charger_discharging_current_set(const struct shell *shell, size_t argc, char **argv)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing current value.");
+		return 0;
+	}
+
+	int err = 0;
+	uint16_t discharging_current = (uint16_t)shell_strtoul(argv[1], 0, &err);
+
+	if (err != 0) {
+		shell_error(shell, "Error: current has to be an integer.");
+		return 0;
+	}
+
+	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
+	uint32_t modules_mask;
+	npmx_error_t err_code = npmx_charger_module_get(charger_instance, &modules_mask);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to get charger module status.");
+		return 0;
+	}
+
+	if ((modules_mask & NPMX_CHARGER_MODULE_CHARGER_MASK) != 0) {
+		shell_error(shell, "Error: charger must be disabled to set discharging current.");
+		return 0;
+	}
+
+	err_code = npmx_charger_discharging_current_set(charger_instance, discharging_current);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to get discharging current value.");
+	}
+
+	err_code = npmx_charger_discharging_current_get(charger_instance, &discharging_current);
```

Comment:
Why?

Reply:
Because in `npmx_charger_discharging_current_set`  are some calculations, and when I set 1000 mA it is 1001 mA

Reply:
Then it should be indicated, that the set result may differ due to some calculations, maybe in the help string (string that goes to `SHELL_CMD`). Otherwise, it will be confusing to users, why the set value is different than the one they are trying to set.